File: src/main/java/com/demonwav/mcdev/platform/forge/util/ForgeConstants.java
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2017 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.platform.forge.util;

import org.jetbrains.annotations.NotNull;

public final class ForgeConstants {

    @NotNull public static final String SIDE_ONLY_ANNOTATION = "net.minecraftforge.fml.relauncher.SideOnly";
    @NotNull public static final String SIDE_ANNOTATION = "net.minecraftforge.fml.relauncher.Side";
    @NotNull public static final String SIDED_PROXY_ANNOTATION = "net.minecraftforge.fml.common.SidedProxy";
    @NotNull public static final String MOD_ANNOTATION = "net.minecraftforge.fml.common.Mod";
    @NotNull public static final String EVENT_HANDLER_ANNOTATION = "net.minecraftforge.fml.common.Mod.EventHandler";
    @NotNull public static final String SUBSCRIBE_EVENT_ANNOTATION = "net.minecraftforge.fml.common.eventhandler.SubscribeEvent";
    @NotNull public static final String FML_EVENT = "net.minecraftforge.fml.common.event.FMLEvent";
    @NotNull public static final String EVENT = "net.minecraftforge.fml.common.eventhandler.Event";
    @NotNull public static final String MCMOD_INFO = "mcmod.info";

    private ForgeConstants() {
    }
}

File: src/main/kotlin/com/demonwav/mcdev/platform/forge/util/ForgeConstants.kt
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2019 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.platform.forge.util

object ForgeConstants {

    const val SIDE_ONLY_ANNOTATION = "net.minecraftforge.fml.relauncher.SideOnly"
    const val SIDE_ANNOTATION = "net.minecraftforge.fml.relauncher.Side"
    const val SIDED_PROXY_ANNOTATION = "net.minecraftforge.fml.common.SidedProxy"
    const val MOD_ANNOTATION = "net.minecraftforge.fml.common.Mod"
    const val CORE_MOD_INTERFACE = "net.minecraftforge.fml.relauncher.IFMLLoadingPlugin"
    const val EVENT_HANDLER_ANNOTATION = "net.minecraftforge.fml.common.Mod.EventHandler"
    const val SUBSCRIBE_EVENT_ANNOTATION = "net.minecraftforge.fml.common.eventhandler.SubscribeEvent"
    const val FML_EVENT = "net.minecraftforge.fml.common.event.FMLEvent"
    const val EVENT = "net.minecraftforge.fml.common.eventhandler.Event"
    const val NETWORK_MESSAGE = "net.minecraftforge.fml.common.network.simpleimpl.IMessage"
    const val NETWORK_MESSAGE_HANDLER = "net.minecraftforge.fml.common.network.simpleimpl.IMessageHandler"
    const val MCMOD_INFO = "mcmod.info"
    const val META_INF = "META-INF"
    const val MODS_TOML = "mods.toml"
    const val PACK_MCMETA = "pack.mcmeta"
}
